Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs

```diff
@@ -101,5 +111,78 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util)
             await _dbContext.SaveChangesAsync();
         }
         #endregion
+        #region Test-Question-Selection
+        public async Task<List<QuestionAC>> GetAllTestCategoryQuestionsByIdAsync(int testId, int categoryId)
+        {
+            QuestionAC questionAc = new QuestionAC();
+            List<QuestionAC> questionListAc = new List<QuestionAC>();
+            var testQuestionList = await _dbContext.TestQuestion.Where(x => x.TestId == testId).ToListAsync();
+            var questionList = await _dbContext.Question.Where(x => x.CategoryID == categoryId).Include(y => y.SingleMultipleAnswerQuestion).ThenInclude(x => x.SingleMultipleAnswerQuestionOption).ToListAsync();
+
+            questionList.ForEach(question =>
+            {
+                questionAc = new QuestionAC();
+                questionAc.Question = Mapper.Map<Question, QuestionDetailAC>(question);
+                questionAc.SingleMultipleAnswerQuestion = Mapper.Map<SingleMultipleAnswerQuestion, SingleMultipleAnswerQuestionAC>(question.SingleMultipleAnswerQuestion);
+                if (testQuestionList.Exists(x => x.QuestionId == questionAc.Question.Id && x.TestId == testId))
+                    questionAc.Question.IsSelect = true;
+                questionListAc.Add(questionAc);
+            });
+            return questionListAc;
+        }
+
+        public async Task<string> AddTestQuestionsAsync(List<QuestionAC> questionsToAddTest, int testId)
+        {
+            bool isDeleted = false;
+            TestQuestion testQuestionObj = new TestQuestion();
+            List<TestQuestion> testQuestionList = new List<TestQuestion>();
+            foreach (var questionToAdd in questionsToAddTest)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable 'testQuestionObj'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1854)